Mozillians_Repositioning.md

```diff
@@ -0,0 +1,233 @@
+- user clicks on save button
+- changed fields get sent to API
+- resulting partial profile update gets sent to [profile publisher](#profile-publisher)
+- on success user gets redirected to edit profile view
+
+##### Searching
+
+- user enters something in the search bar on any page
+- while typing the new front-end queries the suggestion API for suggestions
+- Django queries search service according to users permissions for suggestions
+- front-end displays suggestions
+- user hits enter / clicks on search / clicks on a suggested search
+- font-end sends search query to search API
+- Django queries search service according to users permissions
+- new-front end renders search results and updates the search bar to reflect the proper search syntax
+
+##### …
+
+
+### Search Service
+
+The search service will be built on top of elastic search. It should provide the following APIs:
+
+- query for an individual profile by username / user id and return the according profile data
+- a search that exposes most of ES syntax that returns results in one of two formats:
+  - list of usernames / user ids
+  - list of minimal profile data (everything needed to render a card)
+- suggestion / auto-complete for search
+
+The search service will only be accessible by Mozillians, profile publisher and integrity checker.
+Ideally we store raw profile v2 data and just add some prefixed properties if necessary.
+
+#### Profile Updates
+
+The search service will have an endpoint to receive profile updates. This endpoint will insert or
+update the profile it received.
+
+#### Privacy Levels
+
+The search service will be responsible of filtering fields according to privacy settings. We have
+two proposals on how we can achieve this.
+
+1. We use one index per privacy level. This enables simple search but we need to do the filtering on
+   every insertion of data.
+2. We use a single index and include privacy filters in the search query and do filter fields in the
+   result set. This enables simple insertion but requires more work during search time.
+
+
+### Profile Publisher
+
+The profile publisher is the central broker for dealing with profile updates. Its main
+responsibilities are:
+
+- receive user created updates from Mozillians and send them to CIS
+  - the request/respond cycle will wait until the profile was successfully updated in CIS or an
+    error/timeout occurred
+- receive update events from CIS and send profile updates to [search service](#search-service)
+  and [orgchart service](#orgchart-service)
+
+The profile publisher will be implemented in an async mindset. Since it's mainly IO bound this
+will allow us to scale vertically for a long time.
+
+
+### Orgchart Service
+
+The orgchart service will serve any information necessary to render (parts of) the organizational
+structure of Mozilla.
+
+It's to be determined what technology will back this. We will focus on having a stable API.
+
+It exposes the following data profile v2 properties:
```

Comment:
should this API live side by side with person-api as a micro service instead of within mozillians.org codebase?

Reply:
I would move this to an independent code base anyhow. Why would you bundle this with the person-api? For me this is essentially the same than the search service. For me _search service, orgchart service and profile publisher_ are close to person-api.

Reply:
i didnt mean to bundle it with person-api, rather, just to have a similar querying interface
it sounds like to me that we're thinking the same thing, more or less